apps/storefront/src/App.tsx

```diff
@@ -334,6 +334,21 @@ export default function App() {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isOpen]);
 
+  useEffect(() => {
+    const { hash = '' } = window.location;
+
+    const handleHashChange = () => (!hash || hash === '#/') && setOpenPage({ isOpen: false });
+
+    window.addEventListener('hashchange', handleHashChange);
+
+    return () => {
+      window.removeEventListener('hashchange', handleHashChange);
+    };
+    // ignore setOpenPage
+    // due they are funtions that do not depend on any reactive value
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
🤔 what's the harm in honoring `setOpenPage` here? it'll just remount the event listener.

Reply:
I noticed we consistently override `react-hooks/exhaustive-deps`, this is an antipattern. We should be wrapping functions that we don't want to change in `useCallbacks` and adding them to the array. This can cause unexpected issues, like changes not updating the UI and incorrect function/values being executed/passed around.